### DISCUSSION TITLE
slskd port forwarding with ProtonVPN

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [qdm12/gluetun#1966] [Q&A]
Hi,

did anyone managed to used protonvpn's port forwarding with slskd ? 

As the forwarded port changes pretty often, I think that it requires a tool to update slskd's configuration all the time.

**Comment by GordonFreemanK** [comment]
Found this thread while looking for the solution as well.
As far as I know slskd doesn't have an API that allows updating the port. If that assumption is correct, even if something existed that would read the gluetun-provided port, the only thing it could do is update the config file and reload slskd, which is disruptive (as it will fail all downloads).
I think the first thing to do would be to open a ticket with slskd to allow updating the port without restarting, ideally through an API endpoint - you could then call that with https://github.com/afiestas/gluetun-sync

Edit: found a feature request on slskd to that effect: https://github.com/slskd/slskd/issues/1011

**Comment by tryfan** [comment]
You can use:

VPN_PORT_FORWARDING_UP_COMMAND=/bin/sh -c 'sed -i "s/^\([[:space:]]*listen_port:\).*/\1 $(cat /gluetun/forwarded_port)/" /slskddata/slskd.yml'

You have to mount the volume or at least the slskd.yml file to the gluetun container to have it modify the file. The file updates runtime automagically.

**Reply by lispcat** [reply]
I'm using  `VPN_PORT_FORWARDING_UP_COMMAND=/bin/sh -c 'yq -i ".soulseek.listen_port = {{PORT}}" /path/to/your/slskd.yml'`, but note that it depends on the `yq` program.

Gluetun will automatically substitute `{{PORT}}` with the vpn's forwarded port on every invocation of the up command.

**Reply by tryfan** [reply]
You may want to check out gluetun and slskd's updated docs.  There's an integrations section of the config to query gluetun directly for the forwarded port.  Makes the whole setup much cleaner.  My config was actually kinda busted until I fixed it.

**Reply by lispcat** [reply]
Yeah that looked quite promising. I'm currently using slskdN, a fork of slskd, and they don't have vpn integration sadly, aside from their own implementation, which seemed very hacky and not viable given my very fragile and obscure setup, so I'll continue using the current env var method.
